File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    count = 1
    start = section[0]
    
    for i in range(len(section)-1):
        if start + m > section[i+1]:
            continue
        else:
            start = section[i+1]
            count += 1
    
    return count

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.swift
import Foundation

// n: 벽의 개수, m: 롤러의 길이
// 출력: 페인트 칠 횟수
func solution(_ n:Int, _ m:Int, _ section:[Int]) -> Int {
    var result = 0
    var beforePaintedWall = 0
    
    for currentWall in section {
        if beforePaintedWall >= currentWall {
            continue
        }
        
        result += 1
        beforePaintedWall = currentWall + m - 1
    }
    
    return result
}